### ISSUE TITLE
Default provider?

### ISSUE THREAD

**Issue by itstrueimryan** [sfelix-martins/passport-multiauth#89]
Any way to have a default provider used if the `providers` param is not included in the `/oauth/token` request?  Thanks.

**Comment by sfelix-martins** [comment]
@itstrueimryan I think that we can create a config to set a default value or throws the exception. A pull request would be welcome.

**Comment by tumainimosha** [comment]
I had similar need on my project. I approached this by extending the AddCustomerProvider middleware and add a default provider if none is supplied.

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use SMartins\PassportMultiauth\Http\Middleware\AddCustomProvider as BaseProvider;

class AddCustomProvider extends BaseProvider
{
    /**
     * Handle an incoming request.
     *
     * @param  \Illuminate\Http\Request $request
     * @param  \Closure $next
     * @return mixed
     * @throws \League\OAuth2\Server\Exception\OAuthServerException
     */
    public function handle(Request $request, Closure $next)
    {
        /**
         * Set default provider if none selected.
         */
        $defaultGuard = config('auth.defaults.guard');
        $defaultProvider = config("auth.guards.$defaultGuard.provider");

        $provider = $request->input('provider', $defaultProvider);

        $request->request->add(['provider' => $provider]);

        return parent::handle($request, $next);
    }
}
```

In my Http Kernel I replaced AddCustomProvider middleware from this package with my own

```php
protected $routeMiddleware = [
    ...
    'oauth.providers' => \App\Http\Middleware\AddCustomProvider::class,
]
```

**Comment by sfelix-martins** [comment]
Closes #88
